client/blocks/props-viewer/index.jsx

```diff
@@ -0,0 +1,208 @@
+/**
+ * External Dependencies
+ */
+import React, { PropTypes, PureComponent } from 'react';
+
+/**
+ * Internal Dependencies
+ */
+import Card from 'components/card';
+import Gridicon from 'components/gridicon';
+
+// todo: move this file!
+const components = require( '../../../server/devdocs/proptypes-index.json' );
+
+/**
+ * Finds a non-example component in the library of components
+ * @param {string} slug The slug to search for
+ * @return {Array} An array of component matches
+ */
+export function findRealComponent( slug ) {
+	// remove the last character. As of right now, all plural display names are with just an 's'
+	const singular = slug.slice( 0, -1 );
+	return components.filter( ( component ) => {
+		return ( slug === component.slug || singular === component.slug ) && component.includePath.indexOf( 'example' ) < 0;
+	} );
+}
+
+/**
+ * Renders a table of prop-types for auto-documentation
+ */
+class PropsViewer extends PureComponent {
+	static displayName = 'PropsViewer';
```

Comment:
Yes, React will do that. Until #10343 gets merged in, the index-generator can't read a class without `displayName` set. Originally the unit tests read this file, so this can be removed now.